src/variant.h

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2017-2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_VARIANT_H
+#define BITCOIN_VARIANT_H
+
+#include <utility>
+
+#include <boost/variant.hpp>
+
+//! Substitute for C++17 std::variant
+template <class... Types>
+class Variant
+{
+private:
+    boost::variant<Types...> m_var;
+
+public:
+    Variant()
+    {
+        m_var = {};
+    }
+    Variant(const Variant& other)
+    {
+        m_var = other.m_var;
+    }
+    Variant(Variant& other)
+    {
+        m_var = other.m_var;
+    }
+    Variant(Variant&& other)
+    {
+        m_var = std::move(other.m_var);
+    }
+    template <class T>
+    Variant(const T& value)
+    {
+        m_var = value;
+    }
+    template <class T>
+    Variant(T&& value)
+    {
+        m_var = std::move(value);
```

Comment:
In commit "Add a new Variant type" (85882c01d20add49f7b1f8b741e078968ec8080b)

Maybe want `std::forward<T>(value)` instead of `std::move(value)` because this constructor (because it's a template) can be called with lvalues where it would be surprising to move from.

Reply:
If you turn this into an `std::forward` I think you can drop all other copy constructors.

Reply:
I didn't really look into `std::forward` before, I now think I understand how and why it is useful, pretty cool. Thanks.

Reply:
Done